backend/elasticsearch/src/test/java/org/hibernate/search/backend/elasticsearch/orchestration/impl/ElasticsearchSerialWorkProcessorTest.java

```diff
@@ -110,7 +111,51 @@ public void simple_multipleWorksInWorkSet() {
 		replayAll();
 		CompletableFuture<Void> futureAll = processor.endBatch();
 		verifyAll();
-		assertThat( futureAll ).isSameAs( sequenceFuture );
+		assertThat( futureAll ).isPending();
+
+		resetAll();
+		sequenceFuture.complete( null );
+		replayAll();
+		assertThat( futureAll ).isSuccessful();
+	}
+
+	@Test
+	public void simple_sequenceFailure() {
+		ElasticsearchWork<Object> work = work( 1 );
+
+		CompletableFuture<Void> sequenceFuture = new CompletableFuture<>();
+
+		replayAll();
+		ElasticsearchSerialWorkProcessor processor =
+				new ElasticsearchSerialWorkProcessor( sequenceBuilderMock, bulkerMock );
+		verifyAll();
```

Comment:
I think there is no mock to verify. Maybe we can remove it.

Reply:
There are mocks involved just below. Calling `replayAll` and `verifyAll` allows us to check that no method is called on those mocks.
Even if we didn't care, mocks simply shouldn't be used by the processor in their default "record" state: at best it won't do anything, at worst it will mess with the expectations of the next assertions.